plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemTrim.java

```diff
@@ -0,0 +1,79 @@
+package com.denizenscript.denizen.objects.properties.item;
+
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.utilities.Utilities;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+import org.bukkit.Registry;
+import org.bukkit.inventory.meta.ArmorMeta;
+import org.bukkit.inventory.meta.trim.ArmorTrim;
+import org.bukkit.inventory.meta.trim.TrimMaterial;
+import org.bukkit.inventory.meta.trim.TrimPattern;
+
+public class ItemTrim extends ItemProperty<MapTag> {
+
+    // <--[property]
+    // @object ItemTag
+    // @name trim
+    // @input MapTag
+    // @description
+    // Controls the trim on armor.
+    // Allowed keys: material, pattern.
+    // Valid material inputs can be found here: <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/meta/trim/TrimMaterial.html>
+    // Valid pattern inputs can be found here: <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/meta/trim/TrimPattern.html>
+    // Valid pattern and material inputs also include ones added by datapacks, plugins, etc. as a namespaced key.
+    // If an item already has a trim, you can omit either material or pattern to keep the original data while also changing the other option.
+    // For example, if you only want to change the pattern and not the material, you can omit the material, and it will use the already existing material.

```

Comment:
Should specify this is only relevant to the mechanism, to avoid potential confusion when using the tag - just a `For the mechanism:` prefix is fine for now